classes/plugins/PluginRegistry.php

```diff
@@ -202,29 +202,50 @@ private static function instantiatePlugin(string $category, string $pluginName,
         if (!preg_match('/^[a-z0-9]+$/i', $pluginName)) {
             throw new Exception("Invalid product name \"{$pluginName}\"");
         }
+        
+        $plugin = null;
 
         // First, try a namespaced class name matching the installation directory.
         $pluginClassName = "\\APP\\plugins\\{$category}\\{$pluginName}\\" . ucfirst($pluginName) . 'Plugin';
         try {
-            $plugin = class_exists($pluginClassName)
-                ? new $pluginClassName()
-                : static::deprecatedInstantiatePlugin($category, $pluginName);
+            if (class_exists($pluginClassName)) {
+                $plugin = new $pluginClassName(); /** @var Plugin $plugin */
+            } else {
+                $pluginIndexPath = app()->basePath()
+                    . DIRECTORY_SEPARATOR
+                    . 'plugins'
+                    . DIRECTORY_SEPARATOR
+                    . "{$category}"
+                    . DIRECTORY_SEPARATOR
+                    . "{$pluginName}"
+                    . DIRECTORY_SEPARATOR
+                    . 'index.php';
+
+                if (!file_exists($pluginIndexPath)) {
+                    throw new Exception("fallback to plugin wrapper failed as not found at path {$pluginIndexPath} for plugin {$category}/{$pluginName}");
+                }
+
```

Comment:
This whole block is already wrapped in a `try`, so this error (or the `Error` that `require` throws just below) would both be caught and dealt with below anyway. I'd suggest just letting the `try`/`catch` handle a failure from `require`, and removing this check.